packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts

```diff
@@ -428,37 +428,41 @@ export class AggregatedAttestationPool {
     const metrics = opPool.aggregatedAttestationPool;
     const allSlots = Array.from(this.attestationGroupByIndexByDataHexBySlot.keys());
 
-    // always record the previous slot because the current slot may not be finished yet, we may receive more attestations
-    if (allSlots.length > 1) {
-      // last item is current slot, we want the previous one
-      const previousSlot = allSlots.at(-2);
-      if (previousSlot == null) {
-        // only happen right after we start the node
-        return;
-      }
+    // last item is current slot, we want the previous one, if available.
+    const previousSlot = allSlots.length > 1 ? (allSlots.at(-2) ?? null) : null;
+
+    let attestationCount = 0;
+    let attestationDataCount = 0;
 
+    // always record the previous slot because the current slot may not be finished yet, we may receive more attestations
+    if (previousSlot !== null) {
       const groupByIndexByDataHex = this.attestationGroupByIndexByDataHexBySlot.get(previousSlot);
       if (groupByIndexByDataHex != null) {
         metrics.attDataPerSlot.set(groupByIndexByDataHex.size);
 
         let maxAttestations = 0;
         let committeeCount = 0;
         for (const groupByIndex of groupByIndexByDataHex.values()) {
+          attestationDataCount += groupByIndex.size;
           for (const group of groupByIndex.values()) {
-            const attestationCount = group.getAttestationCount();
-            maxAttestations = Math.max(maxAttestations, attestationCount);
-            metrics.attestationsPerCommittee.observe(attestationCount);
+            const attestationCountInGroup = group.getAttestationCount();
+            maxAttestations = Math.max(maxAttestations, attestationCountInGroup);
+            metrics.attestationsPerCommittee.observe(attestationCountInGroup);
             committeeCount += 1;
+
+            attestationCount += attestationCountInGroup;
           }
         }
         metrics.maxAttestationsPerCommittee.set(maxAttestations);
         metrics.committeesPerSlot.set(committeeCount);
       }
     }
 
-    let attestationCount = 0;
-    let attestationDataCount = 0;
-    for (const attestationGroupByIndexByDataHex of this.attestationGroupByIndexByDataHexBySlot.values()) {
+    for (const [slot, attestationGroupByIndexByDataHex] of this.attestationGroupByIndexByDataHexBySlot) {
+      // We have already updated attestationDataCount and attestationCount when looping over `previousSlot`
+      if (slot === previousSlot) {
```

Comment:
this `attetationCount` and `attestationDataCount` is not per slot, it's the total value for the whole pool so should not skip it for this slot